tests/test_collection.py

```diff
@@ -820,3 +820,19 @@ def test_from_items_with_providers(sample_item_collection: ItemCollection) -> No
 
     provider = collection.providers[0]
     assert provider and provider.name == "pystac"
+
+
+def test_from_dict_null_extent(collection: Collection) -> None:
+    # https://github.com/stac-utils/pystac/issues/1558
+    # https://github.com/EOPF-Sample-Service/eopf-stac/issues/18
+    d = collection.to_dict()
+    d["extent"] = None
+    with pytest.warns(UserWarning):
+        Collection.from_dict(d)
```

Comment:
Add an assertion to prove that the extent gets set to global, all time?
```suggestion
        c = Collection.from_dict(d)
        assert c.extent.spatial.to_dict()["bbox"] == [-90, -180, 90, 180]
        assert c.extent.temporal.to_dict()["intervals"] == [None, None]
```